src/toolkit/res_buf.h

```diff
@@ -85,6 +89,13 @@ class ResBuf {
     cap_ = cap;
   }
 
+  /// Sets whether the buffer should keep packaged resources
+  void keep_packaging(bool keep_packaging) {
+    keep_packaging_ = keep_packaging;
```

Comment:
I think we need to test here, too.  We could put the test only here, and then use this method to set this variable in the constructor.

Reply:
we don't use the `capacity` method to set the capacity variable though

Reply:
I did add a test though

Reply:
Maybe we should??

Reply:
Is it every possible at initialization to have an inconsistency in capacity?

Reply:
Currently no, though it probably shouldn't allow a zero or negative capacity

Reply:
I realize that when I said we should "test", here that may not have been clear.  Whether we change the constructor or not, we need to make sure that if we change `keep_packaging_` to a value that is incompatible with `is_bulk_` in this function that we catch it.

Reply:
Ah, yes yes